src/ClientBase.php

```
@@ -268,4 +378,52 @@ protected function buildClient(array &$options) {
         unset($options['handler']);
         return $client;
     }
+
+    /**
+     * Helper function to extract the option codes.
+     *
+     * @param array $options
+     *  The options array.
+     *
+     * @return array
+     *  The list of options with the option-code.
+     */
+    protected function parseOptionCodes(array $options)
+    {
+        $valid_options = [];
+        foreach ($options['option_codes'] as $optionCode) {
+            if (!in_array(strtoupper($optionCode), self::getOptionCodes())) {
+                break;
```

I know this was an already existing method, but I'm thinking maybe we should be throwing an `InvalidArgumentException` exception instead of silently ignoring the code. Available option codes are limited and well-defined and if there's a code provided that is not supported it's most likely an error that the developer should be aware of.